ReactAndroid/src/main/java/com/facebook/react/modules/toast/ToastModule.java

```diff
@@ -46,7 +56,30 @@ public String getName() {
   }
 
   @ReactMethod
-  public void show(String message, int duration) {
-    Toast.makeText(getReactApplicationContext(), message, duration).show();
+  public void show(String message, int duration, final int id) {
+    Toast toast = Toast.makeText(getReactApplicationContext(), message, duration);
+    toast.show();
+    mToasts.put(id, toast);
+    new Timer().schedule(new TimerTask() {
```

Comment:
I was thinking of a `WeakHashMap` with keys being the Toasts and values being the ids. You'd have to loop over it's `entrySet` to find the Toast for the id but there won't be many Toasts.

There's a real chance the garbage collector will remove entries from the HashMap before you call `dismiss` though: http://stackoverflow.com/questions/3243215/how-to-use-weakreference-in-java-and-android-development

The solution with the timer seems OK.